src/common/core/core.js

```diff
@@ -1411,7 +1437,7 @@ define([
         this.createSet = function (node, name) {
             ensureNode(node, 'node');
             ensureRelationName(name, 'name');
-
+            ensureMongoCompatibleKey(name, 'name', false);
```

Comment:
Why is the `hiddenIsFine` set to false for this an not the others?

Reply:
We do not allow 'hidden' names for sets, and the underscore is already being tested in the ensureRelationName.
On the other hand I will check to make all set related functions and after ensureRelationName checks consistent.